addons/purchase/tests/test_purchase.py

```diff
@@ -559,3 +559,33 @@ def test_purchase_order_multi_company(self):
         po = po_form.save()
 
         self.assertEqual(po.order_line[0].price_unit, 3.0)
+
+    def test_action_view_po_when_product_template_archived(self):
+        """
+        Test to ensure that the purchased_product_qty value remains the same
+        after archiving the product template. Also check that the purchased smart
+        button returns the correct purchase order lines.
+        """
+        po = self.env['purchase.order'].create({
+            'partner_id': self.partner_a.id,
+            'order_line': [
+                Command.create({
+                    'product_id': self.product_a.id,
+                    'product_qty': 10,
+                    'price_unit': 1,
+                }),
+            ],
+        })
+        po.button_confirm()
+        template = po.order_line[0].product_id.product_tmpl_id
```

Comment:
```suggestion
        product_tmpl = self.product_a.product_tmpl_id
```
`template` is a somewhat ambiguous for a variable name